ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-commands.sh

```diff
@@ -252,6 +252,13 @@ required_permissions="""
 \"Microsoft.Storage/storageAccounts/listKeys/action\"
 """
 
+# optional permissions when installing cluster in existing vnet
+required_permissions="""
+\"Microsoft.Network/virtualNetworks/checkIpAddressAvailability/read\",
```

Comment:
The permission is needed by recent PR change, but not for all the previous OCP versions, right?

Reply:
should be related with bug https://issues.redhat.com//browse/OCPBUGS-37442, which has been backported to 4.17. So this permission is required on 4.17+ CAPI install. 
Will file doc bug to request to add the permission on 4.17+.

Reply:
Error happens when creating manifests, this should impact both IPI and UPI. 
But I can update to only apply permission on 4.17+.

Reply:
@jinyunma one more question, does it only affect the installation for existing vnet? If so, do we need a separate env var to control it, so we can skip this permission for IPI install without existing vnet?

Reply:
Good suggestion.
As discussed offline, will read install-config and assign proper permissions based on various config.